io/disk/ssd/nvmetest.py

```diff
@@ -54,6 +56,66 @@ def setUp(self):
         self.format_size = pow(2, int(self.format_size))
         cmd = "nvme id-ns %s | grep 'in use' | awk '{print $2}'" % self.id_ns
         self.lba = process.system_output(cmd, shell=True).strip('\n')
+        self.firmware_url = self.params.get('firmware_url', default='')
+        if 'firmware_upgrade' in str(self.name) and not self.firmware_url:
+            self.skip("firmware url not gien")
```

Comment:
is firmware_upgrade a feature for test if yes i don't think we can skip the test.
As well if already system is updated with latest firmware so i think no need to update that , what you think ?